max_auto_associations_e2e_test.go

```diff
@@ -0,0 +1,598 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+var initialMetadataList = [][]byte{{1}, {2}, {3}, {4}, {5}, {6}, {7}, {8}, {9}, {10}}
+
+func createNftHelper(t *testing.T, env *IntegrationTestEnv) TokenID {
+	tokenCreate, err := NewTokenCreateTransaction().
+		SetTokenName("Example Collection").SetTokenSymbol("ABC").
+		SetTokenType(TokenTypeNonFungibleUnique).SetDecimals(0).
+		SetInitialSupply(0).SetMaxSupply(10).
+		SetTreasuryAccountID(env.Client.GetOperatorAccountID()).SetSupplyType(TokenSupplyTypeFinite).
+		SetAdminKey(env.Client.GetOperatorPublicKey()).
+		SetFreezeKey(env.Client.GetOperatorPublicKey()).
+		SetSupplyKey(env.Client.GetOperatorPublicKey()).
+		SetMetadataKey(env.Client.GetOperatorPublicKey()).
+		Execute(env.Client)
+
+	require.NoError(t, err)
+	receipt, err := tokenCreate.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+	return *receipt.TokenID
+}
+
+func createFungibleTokenHelper(decimals uint, t *testing.T, env *IntegrationTestEnv) TokenID {
+	tokenCreate, err := NewTokenCreateTransaction().
+		SetNodeAccountIDs(env.NodeAccountIDs).
+		SetTokenName("ffff").
+		SetTokenSymbol("F").
+		SetTokenMemo("asdf").
+		SetDecimals(decimals).
+		SetInitialSupply(1_000_000).
+		SetTreasuryAccountID(env.Client.GetOperatorAccountID()).
+		SetAdminKey(env.Client.GetOperatorPublicKey()).
+		SetFreezeKey(env.Client.GetOperatorPublicKey()).
+		SetWipeKey(env.Client.GetOperatorPublicKey()).
+		SetSupplyKey(env.Client.GetOperatorPublicKey()).
+		SetFreezeDefault(false).
+		Execute(env.Client)
+	require.NoError(t, err)
+
+	receipt, err := tokenCreate.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	return *receipt.TokenID
+
+}
+
+func createAccountHelper(t *testing.T, env *IntegrationTestEnv, maxAutoAssociations int32) (AccountID, PrivateKey) {
+	newKey, err := PrivateKeyGenerateEd25519()
+	require.NoError(t, err)
+
+	accountCreate1, err := NewAccountCreateTransaction().
+		SetKey(newKey).
+		SetNodeAccountIDs(env.NodeAccountIDs).
+		SetInitialBalance(NewHbar(0)).
+		SetMaxAutomaticTokenAssociations(maxAutoAssociations).
+		Execute(env.Client)
+	require.NoError(t, err)
+
+	receipt, err := accountCreate1.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+	return *receipt.AccountID, newKey
+}
+
+// Limited max auto association tests
+func TestLimitedMaxAutoAssociationsFungibleTokensFlow(t *testing.T) {
+	t.Parallel()
+	env := NewIntegrationTestEnv(t)
+
+	// create token1 with 1 mil supply
```

Comment:
nit: You can leave out the 1 mil supply comment where written. The helper defines that and not the test.

Reply:
Done